File: KeeperFINAL/Models/VaultKeep.cs
namespace KeeperFINAL.Models;
using KeeperFINAL.Interfaces;
using System;

public class VaultKeep : IDbItem<int>
{
  public int Id { get; set; }
  public int VaultId { get; set; }
  public int KeepId { get; set; }
  public string CreatorId { get; set; }
  public DateTime CreatedAt { get; set; }
  public DateTime UpdatedAt { get; set; }
}

// NOTE vaultkeep/keptkeep = fave/faverecipe
public class KeptKeep : Keep
{
  public int KeepId { get; set; }
  public int VaultId { get; set; }
  public int VaultKeepId { get; set; }
  public string VaultKeepCreatorId { get; set; }
}

File: KeeperFINAL.client/src/models/VaultKeep.js
import { Account } from "./Account.js"
export class VaultKeep {
  constructor(data) {
    this.id = data.id
    this.creatorId = data.creatorId
    this.keepId = data.keepId
    this.vaultId = data.vaultId
    this.creator = new Account(data.creator)
    this.createdAt = data.createdAt
    this.updatedAt = data.updatedAt
  }
}